### ISSUE TITLE
Create new transformation chain object

### ISSUE THREAD

**Issue by jessemapel** [USGS-Astrogeology/ale#128]
This object should represent a sequential transformation chain and be able to do the following:

- Provide the rotation between any two reference frames
- Provide the NAIF ID & name of each reference frame
- Provide the NAIF ID & name of any bodies associated with each reference frame
- Provide the type of each reference frame

**Comment by jessemapel** [comment]
Rough design outline for this. Three classes:

* **FrameNode** : This represents a reference frame in the frame chain
* **Rotation** : This represents a rotation between two reference frames
* **RotationChain** : This represents a sequence of rotations that transform between multiple reference frames

The drivers will return FrameNodes and the FrameNodes will return RotationChains when you ask for a rotation to another FrameNode.  I.E.

```
#list of FrameNodes starting with the sensor reference frame, going through J2000, and ending with the target body reference frame
frame_chain = driver.frame_chain

sensor_frame = frame_chain[0]
target_frame = frame_chain[-1]

#FrameChain object representing the rotations from the sensor frame to the target body frame
sensor_to_target_chain = sensor_frame.rotation_to(target_frame)

#Get the entire rotation chain as a single rotation
sensor_to_target = sensor_to_target_chain.simplify()

if sensor_to_target.dynamic:
    #Get the rotations, angular velocities, and times for the time dependent rotation
    quaternions, angular_velocities, rotation_times = sensor_to_target.rotation()
else:
    #Get the constant rotation as a quaternion
    quaternion = sensor_to_target.rotation()
```

**Comment by jessemapel** [comment]
@jlaura @Kelvinrr @kberryUSGS @SgStapleton @scsides @ladoramkershner thoughts on this? It's not trivial to implement, but not too challenging either. You can see the vispy stuff [here](http://vispy.org/scene.html#module-vispy.scene.node)

**Comment by jlaura** [comment]
Closed by #162.
